File: 69-sqrtx/sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if (x<2){
            return x;
        }
        int l = 1;
        int r = x-1;
        int ans = 0;
        while(l <= r){
            int mid = l + (r - l)/2;
            long s = (long) mid * mid;
            if (s == x){
                return mid ;
            } else if (s < x){
                ans = mid;
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return ans;
    }
}

File: arrays /69-sqrtx/sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int s = 0;
        int e = x;
        long long int ans = -1;
        while(s<=e){
            long long mid = s + (e-s)/2;
            long long sq = mid*mid;
            if(sq == x){
                return mid;
            } else if(sq < x){
                ans = mid;
                s = mid + 1;
            } else {
                e = mid - 1;
            }
        } 
    return ans;
    }
};